pynamodb/attributes.py

```diff
@@ -311,6 +314,15 @@ def _set_attributes(self, **attributes: Attribute) -> None:
                 raise ValueError("Attribute {} specified does not exist".format(attr_name))
             setattr(self, attr_name, attr_value)
 
+    def _deserialize(self, attribute_values: Dict[str, Dict[str, Any]]) -> None:
+        """
+        Sets attributes sent back from DynamoDB on this object
+        """
+        for name, attr in self.get_attributes().items():
+            value = attribute_values.get(attr.attr_name, {NULL: True})
+            value = None if NULL in value else attr.deserialize(attr.get_value(value))
```

Comment:
this is now where we skip any attributes that serialized to NULL (or had NULL set on them outside of pynamo)
still not my favorite work-around but i guess no worse than what we currently do